src/models/application.ts

```diff
@@ -1179,19 +1188,16 @@ const getApplicationModel = function (
 							is_invalidated: false,
 							status: 'success',
 						},
-						$orderby: 'created_at desc',
+						$orderby: { created_at: 'desc' },
 					},
 				},
 			} as const;
 
-			const application = (await exports.get(
-				slugOrUuidOrId,
-				appOptions,
-			)) as PineTypedResult<Application, typeof appOptions>;
-			const body: SubmitBody<Application> = {
+			const application = await exports.get(slugOrUuidOrId, appOptions);
+			const body: Partial<Application['Write']> = {
 				should_track_latest_release: true,
 			};
-			const latestRelease = application.owns__release[0];
+			const latestRelease = application.owns__release?.[0];
```

Comment:
Why would `owns__release` come back undefined? Wouldn't it come back as an empty array at worst?

Reply:
No according to the api typings this can be an `Array<Release['Read']>` or not-existant (notice the `?` on the typing here): https://github.com/balena-io/balena-sdk/pull/1551/files#diff-fdfc0089fcbfd0be6aa662d8bffad07eddc414657cb2a66fd6f742b1752099e3R341

Reply:
> No according to the api typings this can be an Array<Release['Read']> or not-existant (notice the ? on the typing here): https://github.com/balena-io/balena-sdk/pull/1551/files#diff-fdfc0089fcbfd0be6aa662d8bffad07eddc414657cb2a66fd6f742b1752099e3R341

I would say that it's now required b/c `as PineTypedResult<Application, typeof appOptions>;` was removed.

I would prefer it if we did v and avoid the `?.` as Matt says
````
const application = (await exports.get(slugOrUuidOrId, appOptions)) as NonNullable<
	OptionsToResponse<
		Application['Read'],
		typeof appOptions,
		typeof slugOrUuidOrId
	>
>
```
which I guess this is going to go away w/ your other PR that fully types

Reply:
There you have it then folks :) I could avoid the `?` and no `PineTypedResult` as the entire function call now is typed